### PR TITLE
Reorder histogram template parameters

### PR DESCRIPTION
**Pull request by danhoeflinger**
Reorder histogram template parameters and remove unused default template argument.

One option for specification of histogram (version 1a from email).

### CODE REVIEW

File: include/oneapi/dpl/pstl/histogram_extension_defs.h
Comment (akukanov) [inline include/oneapi/dpl/pstl/histogram_extension_defs.h:1]: The change is fine if we consider the use of explicit template arguments as invalid/not allowed. If template arguments can be set explicitly, then it is a breaking change.

**Review by mmichel11 — APPROVED** [review]
LGTM

**Comment by adamfidel** [comment]
Based on the discussion today, LGTM.

**Review by adamfidel — APPROVED** [review]

File: include/oneapi/dpl/pstl/histogram_extension_defs.h
Comment (danhoeflinger) [reply include/oneapi/dpl/pstl/histogram_extension_defs.h:1]: After the discussion offline today, it is my understanding that we are going to view this as not a breaking change, but rather a bug fix, considering there is a very small likelihood that any user code explicitly specifies template parameters in code out in the wild.
If such an issue arises, our suggestion will be to remove the explicit template arguments, and if they are used to force a conversion to instead add an explicit conversion to the calling code.

Comment (akukanov) [reply include/oneapi/dpl/pstl/histogram_extension_defs.h:1]: Agreed.

**Pull request merged**